drivers/watchdog/wdt_w91.c

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright (c) 2024 Telink Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/watchdog.h>
+#include <ipc/ipc_based_driver.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+#define DT_DRV_COMPAT telink_w91_watchdog
+
+#define LOG_MODULE_NAME watchdog_w91
+#if defined(CONFIG_WDT_LOG_LEVEL)
+#define LOG_LEVEL CONFIG_WDT_LOG_LEVEL
+#else
+#define LOG_LEVEL LOG_LEVEL_NONE
+#endif
+
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+struct wdt_w91_config_t {
+	uint8_t instance_id;    /* instance id */
+};
+
+struct wdt_w91_data_t {
+	struct ipc_based_driver ipc;   /* ipc driver part */
+	struct wdt_timeout_cfg timeout_config;
+	uint64_t timeout_ms;
+	bool timeout_installed;
+	bool is_enabled;
+};
+
+/* Type defines for the watchdog functions IPC API */
+enum {
+	IPC_DISPATCHER_WATCHDOG_START = IPC_DISPATCHER_WATCHDOG,
+	IPC_DISPATCHER_WATCHDOG_STOP,
+	IPC_DISPATCHER_WATCHDOG_RESET,
+	IPC_DISPATCHER_WATCHDOG_EXPIRE_NOW,
+};
+
+struct watchdog_start_req_t {
```

Comment:
Please, Do not rename standard types.

Reply:
just changed types

Reply:
Github is automatically tracking changes of mentioned lines. There is no changes here, instead of changes just your comment that its changed...

Reply:
Sorry, I missed exactly this type but changed others